obsplus/structures/grid.py

```diff
@@ -0,0 +1,1323 @@
+        "gouraud" to apply Gouraud shading (see matplotlib docs)
+
+    Returns
+    -------
+    fig : matplotlib Figure
+        Figure containing the resultant plot axes
+    ax : matplotlib Axes
+        Axes containing the resultant plot
+    """
+    figsize = kwargs.pop("figsize", (9, 9))
+    cmap = kwargs.pop("cmap", "rainbow")
+    alpha = kwargs.pop("alpha", 0.5)
+    legend_label = kwargs.pop("legend_label", "Value")
+    shading = kwargs.pop("shading", "flat")
+
+    fig = plt.figure(figsize=figsize)
+
+    # Define the plot grid and create a new axes
+    gs = gridspec.GridSpec(
+        2, 2, width_ratios=[10, 0.5], height_ratios=[10, 0.5], wspace=0.05, hspace=0.05
+    )
+    ax = fig.add_subplot(gs[0])
+
+    # Plot the grid with the appropriate user args
+    cmesh = ax.pcolormesh(
+        x1_coords, x2_coords, values, cmap=cmap, alpha=alpha, shading=shading
+    )
+
+    # Make the plot look nice and add a legend
+    ax.xaxis.tick_top()
+
+    ax2 = fig.add_subplot(gs[2])
+    fig.colorbar(cmesh, cax=ax2, orientation="horizontal", label=legend_label)
+
+    # Make the plot look nice
+    ax.set_aspect("equal")
+    return fig, ax
+
+
+def grid_cross(grid, coord, direction="X"):
+    """
+    Function for returning a profile from a two-dimensional grid
+
+    Parameters
+    ----------
+    grid : Grid
+        Grid from which to pull the values
+    coord : float
+        Coordinate along which to get the profile
+    direction : str
+        Direction along which to get the profile. Possible values are "X"
+        for a profile that parallel to the x-axis and "Y" for a profile
+        that is parallel to the y-axis.
+
+    Returns
+    -------
+    points : np.array
+        List of coordinates along the profile
+    values : np.array
+        Values at each coordinate
+    """
+    if not len(grid.grid_points) == 2:
+        raise TypeError("grid must be a 2D Grid")
+    points = {"X": grid.grid_points[0], "Y": grid.grid_points[1]}
+
+    if (coord < points[direction][0]) and (coord > points[direction][-1]):
+        raise ValueError(
+            f"coord is outside grid bounds: {coord} not in "
+            f"({points[direction][0]}, {points[direction][-1]})"
+        )
+    elif direction == "X":
+        # Slice the grid in the X-direction at that the provided Y-coordinate
+        dummy = points["X"][0]
+        ind = grid.get_index((dummy, coord))[1]
+        values = grid.values[:, ind]
+        return points[direction], values
+    elif direction == "Y":
+        # Slice the grid in the Y-direction at the provided X-coordinate
+        dummy = points["Y"][0]
+        ind = grid.get_index((coord, dummy))[0]
+        values = grid.values[ind, :]
+        return points[direction], values
+    else:
+        raise ValueError(f"Unknown direction: {direction}")
+
+
+# ------------------ Coordinate conversion utilities ----------------- #
+def convert_coords(
```

Comment:
Doesn't need to be done right away but this should be a little more general. For example, the base function could take two (or three) points and return two (or three) points so that Pandas doesn't need to be used.

Reply:
Also, eventually the plan is to put the coordinate stuff in its own module correct?

Reply:
It would definitely be good to put the conversion stuff in it's own module. I had just dumped all of the stuff relating to the Grid in one file for the time being. Where would you envision that living?

> Doesn't need to be done right away but this should be a little more general. For example, the base function could take two (or three) points and return two (or three) points so that Pandas doesn't need to be used.

I just want to make sure we're on the same page, you are suggesting that it should be possible to use some other input than just a pandas DataFrame, and not that pandas shouldn't be used at all, right? I agree that that would be a useful change.